src/screens/TopicPageScreen/index.js

```diff
@@ -142,20 +145,26 @@ const TopicPageScreen = (props) => {
     }
   };
 
+  const handleFollowData = (type) => {
+    if (type === 'sign') {
+      setFollowType('signed');
+      setIsFollow(true);
+    } else if (type === 'anon') {
+      setFollowType('incognito');
+      setIsFollow(true);
+    } else {
+      setFollowType('');
+      setIsFollow(false);
+    }
+  };
+
   const getTopicDetail = async (domain) => {
     try {
-      const query = `?name=${domain}`;
-      const resultGetUserTopic = await getUserTopic(query);
-
-      if (resultGetUserTopic.data) {
-        setIsFollow(true);
-      } else {
-        setIsFollow(false);
-      }
       const resultTopicDetail = await getTopics(domain);
       if (resultTopicDetail.data) {
         const detail = resultTopicDetail.data[0];
         setTopicDetail(detail);
+        handleFollowData(detail.is_followed_by);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `handleFollowData` has been introduced to handle different types of follow actions (signed, anonymous, none). Also, the `getTopicDetail` function has been updated to use this new function. This change improves the modularity and readability of the code.
```diff
+ const handleFollowData = (type) => {
+   if (type === 'sign') {
+     setFollowType('signed');
+     setIsFollow(true);
+   } else if (type === 'anon') {
+     setFollowType('incognito');
+     setIsFollow(true);
+   } else {
+     setFollowType('');
+     setIsFollow(false);
+   }
+ };
+ const getTopicDetail = async (domain) => {
+   try {
+     const resultTopicDetail = await getTopics(domain);
+     if (resultTopicDetail.data) {
+       const detail = resultTopicDetail.data[0];
+       setTopicDetail(detail);
+       handleFollowData(detail.is_followed_by);
+       setMemberCount(Number(detail.followersCount));
+       setIsInitialLoading(false);
+     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->